app/js/components/mixins/ModalEffects.react.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+
+let ModalEffects = InnerComponent => class extends React.Component {
```

Comment:
But I do get your point though. If its the best way to go about this why not. Am just concerned about the re-usability of the code if we happen to have new developers so they won't spend more time in trying to understand the codebase than implementing new features. and to also make it testable